### Data Import

In [102]:
import pandas as pd

data = pd.read_csv('data/2023-06-13-survey.csv')
data.head()

Timestamp     Operating System  CPU Cycle Rate (in GHz)  \
0  6/13/2023 13:21:08              Windows                      3.0   
1  6/13/2023 13:23:33  I use Mac and Linux                   2133.0   
2  6/13/2023 13:24:29                MacOS                      3.2   
3  6/13/2023 13:25:04                MacOS                      3.6   
4  6/13/2023 13:25:31              Windows                      2.3   

   CPU Number of Cores (int)  RAM (in GB)  Hard Drive Size (in GB)  \
0                          8           16                      935   
1                          2            8                      250   
2                          8            8                      256   
3                         12           32                     1000   
4                          8           16                      500   

         GPU (short description as a string)  GPU CUDA Number of Cores (int)  
0  NVIDIA GeForce RTX 2060 with Max-Q Design                            5888  
1                   Intel Iris Plus Graphics                               1  
2                                         M1                               8  
3                                     M2 Max                              38  
4      NVIDIA GeForce RTX 3050 Ti Laptop GPU                            2560

### Exploratory Histograms

In [ ]:
import matplotlib.pyplot as plt
import numpy

l = list(range(1, data.shape[1]))
l.remove(data.columns.get_loc('GPU (short description as a string)')) # remove this column bc it cannot be addressed with a histogram

for i in l:
    plt.title("Feature: {:s}".format(str(data.columns[i])))
    plt.hist(data.iloc[:,i], bins = 25)
    plt.show()

In [100]:

print(data['GPU (short description as a string)'].mode())

data['GPU (short description as a string)'].agg(['count', 'size', 'nunique'])


0    Intel(R) UHD Graphics 620
Name: GPU (short description as a string), dtype: object


count      61
size       62
nunique    48
Name: GPU (short description as a string), dtype: int64

as we can see from above, the GPU is the one category that cannot be addressed with a histogram. Using the methods above, we find very few similarities among responses in the survey. There are 48 different unique responses to the survey for types of GPUs out of 62 total responses. Additionally, the most common response as shown above for this feature is Intel(R) UHD Graphics 620. 